Source/stores.cpp

```diff
@@ -2310,6 +2311,59 @@ void ClearSText(int s, int e)
 	}
 }
 
+void ScrollChatLog(int idx)
+{
+	ClearSText(5, 21);
+	stextup = 5;
+
+	storenumh = ChatLogMessages.size();
+	stextsmax = std::max(storenumh - 4, 0);
+
+	char title[64];
+	sprintf(title, _("Chat History (Messages: %d)"), storenumh);
+	AddSText(0, 1, title, UiFlags::ColorWhitegold | UiFlags::AlignCenter, false);
+
+	for (int l = 5; l < 20; l += 4) {
+		if (idx < ChatLogMessages.size()) {

```

Comment:
Rather than reusing the fairly hacky store UI, this should have its own (simple) rendering similar to https://github.com/diasurgical/devilutionX/blob/master/Source/help.cpp